### PR TITLE
fix cl run call from action-bar

### PR DESCRIPTION
**Pull request by pbhatnagar3**
This fixes cl run command in action-bar. 

Added logic to check if structured_info is present in data.data and if refs is part of data.data.structured_info.

### CODE REVIEW

File: codalab/apps/web/static/js/search/ws_actions.js [hunk 1]
```diff
@@ -98,13 +98,15 @@ var WorksheetActions =  function() {
...
-                                        $(".terminal-output div div:contains(" + k + ")").html(function(idx, html) {
-                                            var link = '/' + references[k]['type'] + 's/' + references[k]['uuid'];
-                                            return html.replace(k, "<a href=" + link + " target='_blank'>" + k + "</a>");
-                                        });
-                                    }, this);
+                                    if (data.data.structured_result && data.data.structured_result.refs){
+                                        var references = data.data.structured_result['refs'];
+                                        Object.keys(references).forEach(function(k) {
+                                            $(".terminal-output div div:contains(" + k + ")").html(function(idx, html) {
+                                                var link = '/' + references[k]['type'] + 's/' + references[k]['uuid'];
```
Comment (percyliang) [inline codalab/apps/web/static/js/search/ws_actions.js]: Do we want to put some more checks here to ensure these are valid?

Comment (pbhatnagar3) [inline codalab/apps/web/static/js/search/ws_actions.js]: I believe that the conditions cover the changes made. And I am working on writing some selenium tests so we can have a better testing framework.

Comment (percyliang) [inline codalab/apps/web/static/js/search/ws_actions.js]: If 'type' and 'uuid' don't exist, or 'type' is not 'bundle' or 'worksheet', we have a problem...

**Comment by pbhatnagar3** [comment]
Yes. I was assuming that structured_info, if available, would have the uuid and type. I have added conditions to take care if that is not the case. Also added conditions to ensure that links are created only when the type is either a bundle or a worksheet. 
 
@percyliang @kashizui please review.

File: codalab/apps/web/static/js/search/ws_actions.js [hunk 2]
```diff
@@ -98,13 +98,18 @@ var WorksheetActions =  function() {
...
-                                    }, this);
+                                    if (data.data.structured_result && data.data.structured_result.refs){
+                                        var references = data.data.structured_result['refs'];
+                                        Object.keys(references).forEach(function(k) {
+                                            $(".terminal-output div div:contains(" + k + ")").html(function(idx, html) {
+                                                var hyperlink_info = references[k];
+                                                if (hyperlink_info.uuid && (hyperlink_info.type === 'bundle' || hyperlink_info.type === 'worksheet')) {
+                                                    var link = '/' + hyperlink_info['type'] + 's/' + hyperlink_info['uuid'];
+                                                    return html.replace(k, "<a href=" + link + " target='_blank'>" + k + "</a>");
+                                                }
```
Comment (percyliang) [inline codalab/apps/web/static/js/search/ws_actions.js]: If there's an error, don't just fail silently!  Log it to some place and also have selenium check this.  Also, remember to fail gracefully (this is not happening now).

**Pull request merged**